src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -205,6 +205,15 @@ Value MathBuilder::floorDiv(Value lhs, Value rhs) const {
   llvm_unreachable("expected int");
 }
 
+// return (lhs * rhs) + acc
+Value MathBuilder::fma(Value lhs, Value rhs, Value acc) const {
+  assert((lhs.getType() == rhs.getType()) && (rhs.getType() == acc.getType()) &&
+         "expected same type");
+  if (lhs.getType().isa<VectorType>())
```

Comment:
Thanks for adding this! `FMAOp` is for floating-point only, so element type should be checked.

Reply:
Added type check logic. Thanks.

Reply:
`isFloatWithVector` is enough.

```c
if (isFloatWithVector(lhs.getType()))
  return b().create<vector::FMAOp>(loc(), lhs, rhs, acc);
return add(mul(lhs, rhs), acc);
```

Reply:
Fixed with small modification. Thanks!